tests/main/prepare-image-gating/task.yaml

```diff
@@ -0,0 +1,27 @@
+summary: Check that prepare-image works for grub-systems
+
+backends: [-autopkgtest]
+
+systems: [-ubuntu-core-*, -fedora-*, -opensuse-*, -arch-*]
+
+environment:
+    ROOT: "$PWD/root"
+    SEED: "$PWD/root/system-seed"
+
+execute: |
+    echo Running prepare-image
+    # TODO: variant were we pass in the gating snap via --snap *.snap
+    cat >custom.json <<EOF
+    {
+      "validation": "enforce"
+    }
+    EOF
+
+    su -c "SNAPPY_USE_STAGING_STORE=$SNAPPY_USE_STAGING_STORE snap prepare-image --customize custom.json $TESTSLIB/assertions/gating-20-amd64.model $ROOT" test
+
+    GATED_REV=2
+
+    echo Verifying the result
+    ls -lR "$SEED"
```

Comment:
Nitpick^2 I guess this is mostly useful for debug, so maybe move it to debug section?

Reply:
Or `find "$SEED" -ls || true` when in debug section.